cmd/gen-go-sample/out.go

```diff
@@ -133,24 +139,50 @@ func writeDefine(txt string, st *symTab, gen *generator) error {
 var fmtStrReplacer = strings.NewReplacer("%%", "%%", "%s", "%v")
 
 func writePrint(pFmt string, pArgs []string, st *symTab, gen *generator) error {
-	var sb strings.Builder
-	for _, arg := range pArgs {
-		sb.WriteString(", ")
-
-		sc, report := initScanner(arg)
-		path, _, err := writePath(sc, st, gen.descInfo)
-		if err != nil {
-			return report(err)
-		}
-		sb.WriteString(path)
+	argList, err := writePaths(pArgs, st, gen)
+	if err != nil {
+		return err
 	}
 
-	gen.pt.Printf("fmt.Printf(%q%s)", fmtStrReplacer.Replace(pFmt)+"\n", sb.String())
+	gen.pt.Printf("fmt.Printf(%q%s)", fmtStrReplacer.Replace(pFmt)+"\n", argList)
 	gen.imports[pbinfo.ImportSpec{Path: "fmt"}] = true
 
 	return nil
 }
 
+func writeDump(fnFmt string, fnArgs []string, contPath string, st *symTab, gen *generator) error {
+	argList, err := writePaths(fnArgs, st, gen)
+	if err != nil {
+		return err
+	}
+
+	fn := fnFmt
+	if len(argList) != 0 {
+		fn = fmt.Sprintf("fmt.Sprintf(%q%s)", fmtStrReplacer.Replace(fnFmt), argList)
+	}
+
+	sc, report := initScanner(contPath)
+	cont, typ, err := writePath(sc, st, gen.descInfo)
+
```

Comment:
can you delete the line break here, bringing the error check next to its source.

Reply:
Done!